doc/specs/#5000 - Process Model 2.0/#1032 - Elevation Quality of Life Improvements.md

```diff
@@ -0,0 +1,385 @@
+identify that the profile should _always_ run elevated. That way, they could
+open the profile from the dropdown menu of an otherwise unelevated window and
+have the elevated window open with the profile automatically.
+
+We'll be adding the `"elevated": true|false` setting as an _optional_
+per-profile setting, with a default value of _not being set_. When set to `true`
+or `false`, we'll check to see if this window is elevated before creating the
+connection for this profile. If the window has a different elevation level than
+requested in the profile, then we'll create a new window with the requested
+elevation level to handle the new connection.
+
+If the user tries to open an `"elevated": true` profile in a window that's
+already elevated, then a new tab/split will open in the existing window, rather
+than spawning an additional elevated window.
+
+There are three situations where we're creating new terminal instances: new
+tabs, new splits, and new windows. Currently, these are all actions that are
+also exposed in the `wt` commandline as subcommands. We can convert from the
+commandline arguments into these actions already. Therefore, it shouldn't be too
+challenging to convert these actions back into the equal commandline arguments.
+
+
+For the following examples, let's assume the user is currently in an unelevated
+Terminal window.
+
+When the user tries to create a new elevated tab, we'll need to create a new
+process, elevated, with the following commandline:
+
+```
+wt new-tab [args...]
+```
+
+When we create this new `wt` instance, it will obey the glomming rules as
+specified in [Session Management Spec]. It might end up glomming to another
+existing window at that elevation level, or possibly create its own window.
+
+Similarly, for a new elevated window, we can make sure to pass the `-s -1` arguments
+to `wt`. These parameters indicate that we definitely want this command to run in a
+new window, regardless of the current glomming settings.
+
+```
+wt -s -1 new-tab [args...]
+```
+
+However, creating a new **split pane** is a little trickier. Invoking the `wt
+split-pane [args...]` is straightforward enough.
+
+⚠ **TODO**: For discussion:
+
+If the current window doesn't have the same elevation level as the
+requested profile, do we always want to just create a new split? If the command
+ends up glomming to an existing window, does that even make sense? That invoking
+an elevated split in an unelevated window would end up splitting the elevated
+window? It's very possible that the user wanted a split in the tab they're
+currently in, in the unelevated window, but they don't want a split in the
+elevated window.
+
+What if there's not space in the elevated window to create the split (but there
+would be in the current window)? That would sure make it seem like nothing
+happened, silently.
+
+We could alternatively have cross-elevation splits default to always opening a

```

Comment:
> mitigate some of the odd
Indeed.

I like the "don't split things cross elevation; do a new tab instead". Splitting is weird when the context switches. Either new tab implicitly or a warning that what you requested couldn't be completed as non-sense do you want to:
A. Make me a new tab instead.
B. Forget it and cancel. I'll go fix my config.

That would be my best idea, at least.

Reply:
(I've added this suggestion, but I'm still leaning towards "open a new tab always")